examples/asr/asr_vad/speech_to_text_with_frame_vad.py

```diff
@@ -0,0 +1,636 @@
+
+import torch
+import yaml
+from omegaconf import DictConfig, OmegaConf
+from torch.profiler import ProfilerActivity, profile, record_function
+from tqdm import tqdm
+
+from nemo.collections.asr.data import feature_to_text_dataset
+from nemo.collections.asr.metrics.rnnt_wer import RNNTDecodingConfig
+from nemo.collections.asr.metrics.wer import CTCDecodingConfig, word_error_rate
+from nemo.collections.asr.models import ASRModel, EncDecClassificationModel
+from nemo.collections.asr.parts.utils.manifest_utils import read_manifest, write_manifest
+from nemo.collections.asr.parts.utils.vad_utils import (
+    extract_audio_features,
+    generate_overlap_vad_seq,
+    generate_vad_segment_table,
+    get_vad_stream_status,
+    init_frame_vad_model,
+)
+from nemo.core.config import hydra_runner
+from nemo.utils import logging
+
+try:
+    from torch.cuda.amp import autocast
+except ImportError:
+
+    @contextlib.contextmanager
+    def autocast(enabled=None):
+        yield
+
+
+@dataclass
+class InferenceConfig:
+    # Required configs
+    asr_model: Optional[str] = None  # Path to a .nemo file or a pretrained NeMo model on NGC
+    vad_model: Optional[str] = None  # Path to a .nemo file or a pretrained NeMo model on NGC
+    vad_config: Optional[str] = None  # Path to a yaml file containing VAD post-processing configs
+    manifest_filepath: Optional[str] = None  # Path to dataset's JSON manifest
+    audio_dir: Optional[str] = None  # Path to a directory containing audio files, use this if no manifest is provided
+
+    use_rttm: bool = True  # whether to use RTTM
+    rttm_mode: str = "mask"  # how to use RTTM files, choices=[`mask`, `drop`]
+    feat_mask_val: Optional[float] = None  # value used to mask features based on RTTM, set None to use defaults
+    normalize: Optional[
+        str
+    ] = "post_norm"  # whether and where to normalize audio feature, choices=[None, `pre_norm`, `post_norm`]
+    normalize_type: str = "per_feature"  # how to determine mean and std used for normalization
+    normalize_audio_db: bool = False  # whether to normalize loudness of audio before extracting audio features
+    normalize_audio_db_target: float = -25.0  # target dB for audio loudness normalization
+
+    profiling: bool = False  # whether to enable pytorch profiling
+
+    # General configs
+    batch_size: int = 1  # batch size for ASR. Feature extraction and VAD only support single sample per batch.
+    num_workers: int = 8
+    sample_rate: int = 16000
+    frame_unit_time_secs: float = 0.01  # unit time per frame in seconds, equal to `window_stride` in ASR configs.
+    audio_type: str = "wav"
+
+    # Output settings, no need to change
+    output_dir: Optional[str] = None  # will be automatically set by the program
+    output_filename: Optional[str] = None  # will be automatically set by the program
+    pred_name_postfix: Optional[str] = None  # If you need to use another model name, rather than standard one.
```

Comment:
LLM models are suggesting:

rather than standard one -> `other` than `the` standard one.